### PR TITLE
Allow multiple zookeeper bindings

### PR DESCRIPTION
**Pull request by ryanhall07**
In addition, split a ZookeeperModule out of the ZkLeaseModule, so it can
be reused by other Zookeeper installs later.

The motivation is to add a connection to a global zk for cluster
weights.

### CODE REVIEW

File: misk-zookeeper/src/main/kotlin/misk/clustering/zookeeper/ZkLeaseModule.kt [hunk 1]
```diff
@@ -9,20 +9,20 @@ import misk.inject.KAbstractModule
...
+ */
+class ZkLeaseModule(private val config: ZookeeperConfig) : KAbstractModule() {
   override fun configure() {
-    bind<ZookeeperConfig>().toInstance(config)
-    multibind<Service>().to<ZkService>()
     multibind<Service>().to<ZkLeaseManager>()
     multibind<Service>().to(RepeatedTaskQueue::class.toKey(ForZkLease::class)).asSingleton()
     bind<LeaseManager>().to<ZkLeaseManager>()
-    bind<CuratorFramework>().toProvider(CuratorFrameworkProvider::class.java).asSingleton()
+    install(ZookeeperModule(config, ForZkLease::class))
```
Comment (wesleyk) [inline misk-zookeeper/src/main/kotlin/misk/clustering/zookeeper/ZkLeaseModule.kt]: nice

File: misk-zookeeper/src/test/kotlin/misk/clustering/zookeeper/ZkTestModule.kt [hunk 2]
```diff
@@ -27,9 +29,10 @@ internal class ZkTestModule : KAbstractModule() {
     val truststrorePath = this::class.java.getResource("/zookeeper/truststore.jks").path
     bind<String>().annotatedWith<AppName>().toInstance("my-app")
 
+
```
Comment (wesleyk) [inline misk-zookeeper/src/test/kotlin/misk/clustering/zookeeper/ZkTestModule.kt]: revert

File: misk-zookeeper/src/main/kotlin/misk/zookeeper/ZookeeperModule.kt [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+import org.apache.curator.framework.CuratorFramework
+import javax.inject.Provider
+import kotlin.reflect.KClass
+
+/**
+ * Binds a [CuratorFramework] for an application to use. A [Service] is also installed to manage
+ * the lifecycle of the [CuratorFramework].
+ *
+ * If an application needs to connect to multiple Zookeepers, an optional qualifier can be passed
+ * resulting in an annotated [CuratorFramework] binding.
```
Comment (wesleyk) [inline misk-zookeeper/src/main/kotlin/misk/zookeeper/ZookeeperModule.kt]: worth pointing to leases as an example?

Comment (wesleyk) [reply misk-zookeeper/src/main/kotlin/misk/zookeeper/ZookeeperModule.kt]: eh the example you have below is already plenty descriptive

File: misk-zookeeper/src/main/kotlin/misk/clustering/zookeeper/ZkService.kt [hunk 4]
```diff
@@ -5,16 +5,13 @@ import com.google.inject.Key
...
 
-@Singleton
-internal class ZkService @Inject internal constructor(
+internal class ZkService internal constructor(
   private val curatorFramework: CuratorFramework
 ) : AbstractIdleService(), DependentService {
 
   override val consumedKeys: Set<Key<*>> = setOf()
-  override val producedKeys: Set<Key<*>> = setOf(ZookeeperModule.serviceKey)
+  override val producedKeys: Set<Key<*>> = setOf(ZkLeaseModule.serviceKey)
```
Comment (mightyguava) [inline misk-zookeeper/src/main/kotlin/misk/clustering/zookeeper/ZkService.kt]: won't this end up in multiple instances of ZkService producing the same key? Might be useful to have a test where you start multiple zookeepers

Comment (ryanhall07) [reply misk-zookeeper/src/main/kotlin/misk/clustering/zookeeper/ZkService.kt]: fixed this up in a new commit 93d007f

I actually went with reusing the same zookeeper instances in tests. Not sure there is a lot of value in actually having separate instances running

**Pull request merged**